### DISCUSSION TITLE
Hello GitHub, there is a Chinese platform that is increasingly bulk copying users and repositories from GitHub

### DISCUSSION THREAD

**Discussion by sionnx** [community/community#129872] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

Hello GitHub, recently there is a Chinese platform (gitcode.com), jointly built by CSDN and Huawei Cloud, which is copying most of the Github repositories and pretending to be official repositories. For example: https://github.com/python -> https://gitcode.com/python, https://github.com/torvalds -> https://gitcode.com/torvalds. More and more users and repositories are being copied to GitCode. Of course, this also includes my github repository at https://gitcode.com/theseann.

**Comment by astro-angelfish** [comment]
~~For users that feel offended by [GitCode](gitcode.com)~~, GitCode has **already** enabled their pay-to-download function to earn profit and **breaking the open source license**, users might need to add some political sensitive content to their repository to avoid getting their repository mirrored as a temporary solution like [this](https://github.com/astro-angelfish/astro-angelfish/blob/main/.anti-gitcode). For Chinese users they could bring up issues for their copyrights in Chinese to tell GitCode to remove their repository directly.

By the way **GitCode could take down reputations of original authors by abusing their username mirrored** as original author has no control on their mirrored account/username. GitCode will also have the ability to launch an attack like [XZ Backdoor Incident](https://nvd.nist.gov/vuln/detail/CVE-2024-3094) as they gained the same reputation as the original author. Thankfully they currently haven't been found mirroring the private repositories.

What I have to say is that they could effectively suppress the Chinese open source community and earn more profit since there is no legal regulations base on open source license in China and there is some difficulties for Chinese user to access github.

**Comment by chenryace** [comment]
# Please sue gitcode!
incluiding CSDN and huawei cloud! we are offended！

**Comment by astro-angelfish** [comment]
I've come up an experiment with [this dictionary](https://github.com/astro-angelfish/astro-angelfish/blob/main/.anti-gitcode) to avoid getting my repository mirrored by gitcode before github formally sues gitcode. But I do not have enough stars to get mirrored anyway.

**Reply by sekedus** [reply]
Also add `tiananmen square 1989`

**Comment by chickun-io** [comment]
As we all know, CSDN is a shorthand for "**C**opy **S**teal pay-**D**ownload **N**ot-work".

**Comment by ebndev** [comment]
Hi @theseann, 👋🏻 We really appreciate you flagging this. The platform team has been aware of this situation and I have forwarded this discussion as well.

 In the future, the best route to get this to the proper GitHub team is to use our [abuse reporting tools](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam). Here's all the info:

- [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam)
- [Reporting a user](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-user)
- [Reporting an issue or pull request](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-an-issue-or-pull-request)
- [Reporting a comment](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-comment)

You can report behavior and content that violates community guidelines and terms. We are going to close this post, but for this and any future incidents, please refer to the links above.

Thank you!

**Comment by miladxandi** [comment]
> ### Select Topic Area
> 
> Show & Tell
> 
> ### Body
> 
> Hello GitHub, recently there is a Chinese platform (gitcode.com), jointly built by CSDN and Huawei Cloud, which is copying most of the Github repositories and pretending to be official repositories. For example: https://github.com/python -> https://gitcode.com/python, https://github.com/torvalds -> https://gitcode.com/torvalds. More and more users and repositories are being copied to GitCode. Of course, this also includes my github repository at https://gitcode.com/theseann.

Chinese are slightly better than you american and european.
